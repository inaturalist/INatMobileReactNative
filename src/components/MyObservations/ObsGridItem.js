// @flow

import DisplayTaxonName from "components/DisplayTaxonName";
import { View } from "components/styledComponents";
import type { Node } from "react";
import React from "react";
import Photo from "realmModels/Photo";

import ObsImagePreview from "./ObsImagePreview";
import ObsUploadStatus from "./ObsUploadStatus";

type Props = {
  observation: Object,
  width?: string,
  height?: string,
  style?: Object,
  uploadStatus: Object
};

const ObsGridItem = ( {
  observation,
  width = "w-full",
  height,
  style,
  uploadStatus
}: Props ): Node => (
  <ObsImagePreview
    source={{
      uri: Photo.displayLocalOrRemoteMediumPhoto(
        observation?.observationPhotos?.[0]?.photo
      )
    }}
    width={width}
    height={height}
    style={style}
    obsPhotosCount={observation?.observationPhotos?.length ?? 0}
    hasSound={!!observation?.observationSounds?.length}
    isMultiplePhotosTop
    testID={`MyObservations.gridItem.${observation.uuid}`}
  >
    <View className="absolute bottom-0 flex p-2 w-full">
      <ObsUploadStatus
        observation={observation}
        uploadStatus={uploadStatus}
        layout="horizontal"
        white
        classNameMargin="mb-1"
      />
      <DisplayTaxonName
        taxon={observation?.taxon}
        scientificNameFirst={
          observation?.user?.prefers_scientific_name_first
        }
        layout="vertical"
        color="text-white"
      />
    </View>
  </ObsImagePreview>
);

export default ObsGridItem;
