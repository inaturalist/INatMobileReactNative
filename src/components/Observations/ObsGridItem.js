// @flow

import classnames from "classnames";
import DisplayTaxonName from "components/DisplayTaxonName";
import { ImageBackground, Pressable, View } from "components/styledComponents";
import { t } from "i18next";
import type { Node } from "react";
import React from "react";
import LinearGradient from "react-native-linear-gradient";
import IconMaterial from "react-native-vector-icons/MaterialIcons";
import Observation from "realmModels/Observation";
import Photo from "realmModels/Photo";
import colors from "styles/tailwindColors";

import ObsStatus from "./ObsStatus";
import UploadButton from "./UploadButton";

type Props = {
  observation: Object,
  handlePress: Function,
  uri?: string,
  height?: string,
  width?: string,
};

const ObsGridItem = ( {
  handlePress,
  observation,
  uri,
  width = "w-full",
  height = "h-[172px]"
}: Props ): Node => {
  const onPress = () => handlePress( observation );

  const photo = observation?.observationPhotos?.[0]?.photo;

  const totalObsPhotos = observation?.observationPhotos?.length;
  const hasMultiplePhotos = totalObsPhotos > 1;
  const hasSound = !!observation?.observationSounds?.length;
  const filterIconName = totalObsPhotos > 9 ? "filter-9-plus" : `filter-${totalObsPhotos || 2}`;

  const imageUri = uri === "project"
    ? Observation.projectUri( observation )
    : { uri: Photo.displayLocalOrRemoteMediumPhoto( photo ) };

  const showStats = () => {
    if ( uri !== "project" && observation.needsSync() ) {
      return (
        <View className="absolute bottom-0 right-0">
          <UploadButton observation={observation} />
        </View>
      );
    }
    return (
      <ObsStatus
        observation={observation}
        layout="horizontal"
        color={colors.white}
      />
    );
  };

  return (
    <Pressable
      onPress={onPress}
      className={classnames( "rounded-[17px] overflow-hidden", height, width )}
      testID={`ObsList.gridItem.${observation.uuid}`}
      accessibilityRole="link"
      accessibilityLabel={t( "Navigate-to-observation-details" )}
    >
      <View className={classnames( "rounded-[17px] overflow-hidden relative w-full", height )}>
        {imageUri && imageUri.uri ? (
          <ImageBackground
            source={imageUri}
            className="grow aspect-square"
            testID="ObsList.photo"
          >
            <LinearGradient
              className="bg-transparent absolute inset-0"
              colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.5) 100%)"]}
            />
          </ImageBackground>
        ) : (
          <LinearGradient
            colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.5) 100%)"]}
          >
            <View className="grow aspect-square" />
          </LinearGradient>
        )}

        <View className={classnames( "z-100 absolute flex justify-between p-2 w-full", height )}>
          <View
            className={classnames( "flex justify-between", {
              "flex-row-reverse": hasMultiplePhotos
            } )}
          >
            {hasMultiplePhotos && (
              <IconMaterial
                // $FlowIgnore
                name={filterIconName}
                color={colors.white}
                size={22}
              />
            )}
            {( hasSound || true ) && (
              <IconMaterial name="volume-up" color={colors.white} size={22} />
            )}
          </View>

          <View>
            {showStats()}
            <DisplayTaxonName
              taxon={observation?.taxon}
              scientificNameFirst={
                observation?.user?.prefers_scientific_name_first
              }
              layout="vertical"
              color="text-white"
            />
          </View>
        </View>
      </View>
    </Pressable>
  );
};

export default ObsGridItem;
