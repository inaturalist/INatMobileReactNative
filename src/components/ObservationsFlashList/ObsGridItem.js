// @flow

import { Body2, DisplayTaxonName } from "components/SharedComponents";
import { View } from "components/styledComponents";
import type { Node } from "react";
import React, { useMemo } from "react";
import Photo from "realmModels/Photo";
import { useCurrentUser, useFontScale } from "sharedHooks";

import ObsImagePreview from "./ObsImagePreview";
import ObsUploadStatus from "./ObsUploadStatus";

type Props = {
  explore: boolean,
  height?: string,
  observation: Object,
  onPress: Function,
  style?: Object,
  uploadProgress?: number,
  width?: string
};

const ObsGridItem = ( {
  explore,
  height = "w-[200px]",
  observation,
  onPress,
  style,
  uploadProgress,
  width = "w-[200px]"
}: Props ): Node => {
  const photo = observation?.observationPhotos?.[0]?.photo
    || observation?.observation_photos?.[0]?.photo
    || null;
  const photoCount = observation?.observationPhotos?.length
    || observation?.observation_photos?.length;
  const hasSound = !!(
    observation?.observationSounds?.length
    || observation?.observation_sounds?.length
  );
  const currentUser = useCurrentUser( );
  const { isLargeFontScale } = useFontScale();

  const displayTaxonName = useMemo( ( ) => (
    <DisplayTaxonName
      bottomTextComponent={Body2}
      color="text-white"
      ellipsizeCommonName
      keyBase={observation?.uuid}
      layout="vertical"
      prefersCommonNames={currentUser?.prefers_common_names}
      scientificNameFirst={currentUser?.prefers_scientific_name_first}
      showOneNameOnly={!explore || isLargeFontScale}
      taxon={observation?.taxon}
    />
  ), [
    currentUser?.prefers_common_names,
    currentUser?.prefers_scientific_name_first,
    explore,
    isLargeFontScale,
    observation?.taxon,
    observation?.uuid
  ] );

  return (
    <ObsImagePreview
      source={{
        uri: Photo.displayLocalOrRemoteMediumPhoto( photo )
      }}
      width={width}
      height={height}
      style={style}
      obsPhotosCount={photoCount ?? 0}
      hasSound={hasSound}
      isMultiplePhotosTop
      testID={`MyObservations.gridItem.${observation.uuid}`}
      useShortGradient={!explore}
      iconicTaxonName={observation.taxon?.iconic_taxon_name}
      white
    >
      <View className="absolute bottom-0 flex p-2 w-full">
        <ObsUploadStatus
          classNameMargin="mb-1"
          explore={explore}
          layout="horizontal"
          observation={observation}
          onPress={onPress}
          progress={uploadProgress}
          white
        />
        {displayTaxonName}
      </View>
    </ObsImagePreview>
  );
};

export default ObsGridItem;
