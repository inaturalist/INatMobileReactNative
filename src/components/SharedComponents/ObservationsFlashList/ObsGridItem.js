// @flow

import { DisplayTaxonName, ObsStatus } from "components/SharedComponents";
import { View } from "components/styledComponents";
import type { Node } from "react";
import React from "react";
import Photo from "realmModels/Photo";
import { useCurrentUser } from "sharedHooks";

import ObsImagePreview from "./ObsImagePreview";
import ObsUploadStatusContainer from "./ObsUploadStatusContainer";

type Props = {
  observation: Object,
  width?: string,
  height?: string,
  style?: Object,
  uploadSingleObservation?: Function,
  uploadState: {
    uploadProgress: Number
  },
  explore: boolean
};

const ObsGridItem = ( {
  observation,
  width = "w-[200px]",
  height = "w-[200px]",
  style,
  uploadSingleObservation,
  uploadState,
  explore
}: Props ): Node => {
  const photoCount = observation?.observationPhotos?.length
    || observation?.observation_photos?.length;
  const currentUser = useCurrentUser( );
  return (
    <ObsImagePreview
      source={{
        uri: Photo.displayLocalOrRemoteMediumPhoto(
          observation?.observationPhotos?.[0]?.photo || observation?.observation_photos?.[0]?.photo
        )
      }}
      width={width}
      height={height}
      style={style}
      obsPhotosCount={photoCount ?? 0}
      hasSound={!!observation?.observationSounds?.length}
      isMultiplePhotosTop
      testID={`MyObservations.gridItem.${observation.uuid}`}
      iconicTaxonName={observation.taxon?.iconic_taxon_name}
      white
    >
      <View className="absolute bottom-0 flex p-2 w-full">
        {explore
          ? (
            <ObsStatus
              observation={observation}
              layout="horizontal"
              testID={`ObsStatus.${observation.uuid}`}
              white
            />
          )
          : (
            <ObsUploadStatusContainer
              observation={observation}
              layout="horizontal"
              white
              classNameMargin="mb-1"
              uploadSingleObservation={uploadSingleObservation}
              uploadState={uploadState}
            />

          )}
        <DisplayTaxonName
          keyBase={observation?.uuid}
          taxon={observation?.taxon}
          scientificNameFirst={
            currentUser?.prefers_scientific_name_first
          }
          layout="vertical"
          color="text-white"
          ellipsizeCommonName
        />
      </View>
    </ObsImagePreview>
  );
};

export default ObsGridItem;
