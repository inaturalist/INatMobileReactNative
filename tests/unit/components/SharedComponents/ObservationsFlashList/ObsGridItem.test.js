import { render, screen } from "@testing-library/react-native";
import ObsGridItem from "components/SharedComponents/ObservationsFlashList/ObsGridItem";
import initI18next from "i18n/initI18next";
import React from "react";
import factory from "tests/factory";

describe( "ObsGridItem", () => {
  beforeAll( async () => {
    await initI18next();
  } );

  describe( "for an observation with a photo", ( ) => {
    const observationWithPhoto = factory( "LocalObservation", {
      observationPhotos: [factory( "LocalObservationPhoto" )]
    } );

    it( "should render", () => {
      const observationWithStablePhotoUrl = factory( "LocalObservation", {
        uuid: "00000000-0000-0000-0000-000000000000",
        observationPhotos: [
          factory( "LocalObservationPhoto", {
            photo: factory( "LocalPhoto", {
              url: "https://inaturalist-open-data.s3.amazonaws.com/photos/1/large.jpeg"
            } )
          } )
        ]
      } );
      render(
        <ObsGridItem
          observation={observationWithStablePhotoUrl}
          uploadState={{ uploadProgress: false }}
        />
      );
      expect( screen ).toMatchSnapshot();
    } );

    it( "should have photo", async ( ) => {
      render(
        <ObsGridItem
          observation={observationWithPhoto}
          uploadState={{ uploadProgress: false }}
        />
      );
      expect( await screen.findByTestId( "ObsList.photo" ) ).toBeTruthy( );
    } );
  } );

  describe( "for an observation without a photo", ( ) => {
    const observationWithoutPhoto = factory( "LocalObservation", {
      uuid: "00000000-0000-0000-0000-000000000000"
    } );

    it( "should render", ( ) => {
      render(
        <ObsGridItem
          observation={observationWithoutPhoto}
          uploadState={{ uploadProgress: false }}
        />
      );
      expect( screen ).toMatchSnapshot();
    } );

    it( "should have an iconic taxon icon", async ( ) => {
      render(
        <ObsGridItem
          observation={observationWithoutPhoto}
          uploadState={{ uploadProgress: false }}
        />
      );
      expect( await screen.findByTestId( "IconicTaxonName.iconicTaxonIcon" ) ).toBeTruthy( );
    } );
  } );
} );
