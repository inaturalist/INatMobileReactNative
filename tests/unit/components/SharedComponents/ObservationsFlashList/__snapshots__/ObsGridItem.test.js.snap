// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObsGridItem for an observation with a photo should render 1`] = `
<View
  style={
    [
      [
        {
          "maxHeight": 210,
        },
        {
          "overflow": "hidden",
        },
        {
          "position": "relative",
        },
        {
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 15,
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
        },
        {
          "width": 200,
        },
        {
          "width": 200,
        },
      ],
    ]
  }
  testID="MyObservations.gridItem.00000000-0000-0000-0000-000000000000"
>
  <View
    style={
      [
        [
          {
            "flexGrow": 1,
          },
          {
            "aspectRatio": 1,
          },
          {
            "position": "relative",
          },
        ],
      ]
    }
  >
    <View
      style={
        [
          [
            {
              "position": "absolute",
            },
            {
              "width": "100%",
            },
            {
              "height": "100%",
            },
          ],
        ]
      }
    >
      <View
        style={
          [
            [
              {
                "flexGrow": 1,
              },
              {
                "aspectRatio": 1,
              },
              {
                "maxHeight": 210,
              },
              {
                "overflow": "hidden",
              },
              {
                "position": "relative",
              },
              {
                "borderBottomLeftRadius": 15,
                "borderBottomRightRadius": 15,
                "borderTopLeftRadius": 15,
                "borderTopRightRadius": 15,
              },
              {
                "width": 200,
              },
              {
                "width": 200,
              },
              {
                "backgroundColor": "#454545",
              },
              {
                "flexShrink": 1,
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "backgroundColor": "#454545",
              },
            ],
          ]
        }
        testID="IconicTaxonName.iconicTaxonIcon"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#BFBFBF33",
                "fontSize": 100,
              },
              undefined,
              {
                "fontFamily": "INatIcon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <Image
      accessibilityIgnoresInvertColors={true}
      source={
        {
          "uri": "https://inaturalist-open-data.s3.amazonaws.com/photos/1/large.jpeg",
        }
      }
      style={
        [
          [
            {
              "flexGrow": 1,
            },
            {
              "aspectRatio": 1,
            },
          ],
        ]
      }
      testID="ObsList.photo"
    />
  </View>
  <BVLinearGradient
    colors={
      [
        0,
        2147483648,
      ]
    }
    endPoint={
      {
        "x": 0,
        "y": 0.75,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      [
        [
          {
            "position": "absolute",
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ],
      ]
    }
  />
  <View
    style={
      [
        [
          {
            "position": "absolute",
          },
          {
            "bottom": 0,
          },
          {
            "display": "flex",
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          {
            "width": "100%",
          },
        ],
      ]
    }
  >
    <View
      accessibilityLabel="Saved observation, in queue to upload"
      accessible={true}
      style={
        [
          [
            {
              "height": 44,
            },
            {
              "justifyContent": "flex-end",
            },
          ],
        ]
      }
    >
      <View>
        <View
          accessibilityLabel="Start upload"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "height": 44,
                "justifyContent": "center",
                "opacity": 1,
                "width": 44,
              },
              false,
              undefined,
            ]
          }
          testID="UploadIcon.start.00000000-0000-0000-0000-000000000000"
        >
          <View
            style={
              [
                [
                  {
                    "position": "relative",
                  },
                  {
                    "position": "relative",
                  },
                  {
                    "left": 0.2,
                  },
                  {
                    "top": 0.1,
                  },
                ],
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 33,
                  },
                  undefined,
                  {
                    "fontFamily": "INatIcon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "fontFamily": "Whitney-Medium",
          },
          [
            [
              {
                "fontSize": 18,
                "lineHeight": 22,
              },
              {
                "fontWeight": "500",
              },
              {
                "color": "#454545",
              },
              [
                {
                  "color": "#ffffff",
                },
              ],
            ],
          ],
        ]
      }
    >
      unknown
    </Text>
  </View>
</View>
`;

exports[`ObsGridItem for an observation without a photo should render 1`] = `
<View
  style={
    [
      [
        {
          "maxHeight": 210,
        },
        {
          "overflow": "hidden",
        },
        {
          "position": "relative",
        },
        {
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 15,
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
        },
        {
          "width": 200,
        },
        {
          "width": 200,
        },
      ],
    ]
  }
  testID="MyObservations.gridItem.00000000-0000-0000-0000-000000000000"
>
  <View
    style={
      [
        [
          {
            "flexGrow": 1,
          },
          {
            "aspectRatio": 1,
          },
          {
            "position": "relative",
          },
        ],
      ]
    }
  >
    <View
      style={
        [
          [
            {
              "position": "absolute",
            },
            {
              "width": "100%",
            },
            {
              "height": "100%",
            },
          ],
        ]
      }
    >
      <View
        style={
          [
            [
              {
                "flexGrow": 1,
              },
              {
                "aspectRatio": 1,
              },
              {
                "maxHeight": 210,
              },
              {
                "overflow": "hidden",
              },
              {
                "position": "relative",
              },
              {
                "borderBottomLeftRadius": 15,
                "borderBottomRightRadius": 15,
                "borderTopLeftRadius": 15,
                "borderTopRightRadius": 15,
              },
              {
                "width": 200,
              },
              {
                "width": 200,
              },
              {
                "backgroundColor": "#454545",
              },
              {
                "flexShrink": 1,
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "backgroundColor": "#454545",
              },
            ],
          ]
        }
        testID="IconicTaxonName.iconicTaxonIcon"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#BFBFBF33",
                "fontSize": 100,
              },
              undefined,
              {
                "fontFamily": "INatIcon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <BVLinearGradient
    colors={
      [
        0,
        2147483648,
      ]
    }
    endPoint={
      {
        "x": 0,
        "y": 0.75,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      [
        [
          {
            "position": "absolute",
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ],
      ]
    }
  />
  <View
    style={
      [
        [
          {
            "position": "absolute",
          },
          {
            "right": 0,
          },
          {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          },
          {
            "top": 0,
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "rgba(244, 239, 244, 1)",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "INatIcon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      [
        [
          {
            "position": "absolute",
          },
          {
            "bottom": 0,
          },
          {
            "display": "flex",
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          {
            "width": "100%",
          },
        ],
      ]
    }
  >
    <View
      accessibilityLabel="Saved observation, in queue to upload"
      accessible={true}
      style={
        [
          [
            {
              "height": 44,
            },
            {
              "justifyContent": "flex-end",
            },
          ],
        ]
      }
    >
      <View>
        <View
          accessibilityLabel="Start upload"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "height": 44,
                "justifyContent": "center",
                "opacity": 1,
                "width": 44,
              },
              false,
              undefined,
            ]
          }
          testID="UploadIcon.start.00000000-0000-0000-0000-000000000000"
        >
          <View
            style={
              [
                [
                  {
                    "position": "relative",
                  },
                  {
                    "position": "relative",
                  },
                  {
                    "left": 0.2,
                  },
                  {
                    "top": 0.1,
                  },
                ],
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 33,
                  },
                  undefined,
                  {
                    "fontFamily": "INatIcon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "fontFamily": "Whitney-Medium",
          },
          [
            [
              {
                "fontSize": 18,
                "lineHeight": 22,
              },
              {
                "fontWeight": "500",
              },
              {
                "color": "#454545",
              },
              [
                {
                  "color": "#ffffff",
                },
              ],
            ],
          ],
        ]
      }
    >
      unknown
    </Text>
  </View>
</View>
`;
